kafka/producer.go

```diff
@@ -17,4 +17,198 @@
 
 package kafka
 
-// TODO
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"go.uber.org/zap"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+	"github.com/twmb/franz-go/plugin/kzap"
+
+	"github.com/elastic/apm-data/model"
+)
+
+// ProducerConfig holds configuration for publishing events to Kafka.
+type ProducerConfig struct {
+	// Broker holds the (host:port) address of the Kafka broker to which
+	// events should be published.
+	Broker string
+
+	// ClientID to use when connecting to Kafka. This is used for logging
+	// and client identification purposes.
+	ClientID string
+	// Version is the software version to use in the Kafka client. This is
+	// useful since it shows up in Kafka metrics and logs.
+	Version string
+
+	// TopicPrefix holds the name of the Kafka topic prefix to which events
+	// should be published. The producer will send messages to 5 different
+	// topics using the <TopicPrefix>.<EventType> format.
+	TopicPrefix string
+
+	// Logger is used for logging producer errors.
+	Logger *zap.Logger
+
+	// Key, if non-nil, is used to set the key for an event message.
+	//
+	// Producer uses murmur2 partitioning: consistent hashing over the
+	// message key, if it is non-nil; random partition assignment if the
+	// message has no key.
+	//
+	// If Key is nil, the message key will be set to the event's trace ID
+	// if there is one, and nil otherwise.
+	Key func(model.APMEvent) []byte
```

Comment:
If we're to implement the record mutators, it wouldn't be necessary to accept this argument and a Key mutator could be passed in instead.

Reply:
Removed!